crypto/types/cache.go

```diff
@@ -0,0 +1,83 @@
+package types
+
+import (
+	lru "github.com/hashicorp/golang-lru/v2"
+)
+
+const (
+	TxHashLen        = 32
+	AddressStringLen = 2 + 20*2
+)
+
+// Cache is a cache of verified signatures
+type Cache struct {
+	data *lru.Cache[string, []byte]
+}
+
+// NewSignatureCache initializes the signature cache.
+// signature verification is one of the most expensive parts in verification
+// by caching it we avoid needing to verify the same signature multiple times
+func NewSignatureCache() *Cache {
+	// 500 * (32 + 42) = 37.5KB
+	cache, err := lru.New[string, []byte](500)
+	if err != nil {
+		panic(err)
+	}
+
+	return &Cache{data: cache}
+}
+
+// Get returns the cached signature if it exists
+func (c *Cache) Get(key string) ([]byte, bool) {
+	if !c.validate(key) {
+		return nil, false
+	}
+
+	return c.data.Get(key)
+}
+
+// Add adds a signature to the cache
+func (c *Cache) Add(key string, value []byte) {
+	// validate
+	if !c.validate(key) {
+		return
+	}
+	// add cache
+	c.data.Add(key, value)
```

Comment:
nit: superfluous comments ;-) 

```suggestion
	if !c.validate(key) {
		return
	}

	c.data.Add(key, value)
```